### DISCUSSION TITLE
Output to the histogram only black colors

### DISCUSSION THREAD

**Discussion by b1zar** [ImageMagick/ImageMagick#2849] [Help]
Hey. Tell me how to implement output to the histogram only certain colors, or rather dark shades from # 000000 to # 050505. So I understand this will have to be done through the fill, but how to specify the range of colors in the fill?

**Comment by fmw42** [comment]
Here is one way in Unix syntax.  Clone the image and get the grayscale (intensity) version and threshold it as desired. In the following, I threshold to make white anything below 50% so only those will become opaque. The pixels above 50% intensity will become transparent. Then put that into the alpha channel of the image. Then output that to txt: format. Then pipe that to grep to keep everything **but** fully transparent. Then pipe that back to convert and create your histogram.


Compare these two output histograms.

```
convert rose: txt:- | \
convert - -alpha off -format "%c" histogram:info:

```
and

```
convert rose:  \
\( +clone -colorspace gray -threshold 50% -negate \) \
-alpha off -compose copy_opacity -composite \
-background black -alpha background -type truecoloralpha txt:- | \
grep -v "none" | \
convert - -alpha off -format "%c" histogram:info:

```
